lib/puppet.rb

```diff
@@ -74,7 +74,7 @@ class << self
       end
 
       if locale_path
-        if Gem.loaded_specs['gettext-setup'].version < Gem::Version.new('0.8')
+        if GettextSetup.method(:initialize).parameters.count == 1
```

Comment:
That won't work here because both arguments are optional. `GettextSetup.method(:initialize).arity` returns `-1`.